### PR TITLE
Adds Dockerfile to create and application.properties to connect to an empty mysql database. Also adds basic tables.

### PR DESCRIPTION
**Pull request by TheLastKiwi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,18 @@
 All the reads
 
 All code should have JavaDoc, unit tests.
+
+
+To start your Docker container with a mysql database, do the following:
+0. Install and start Docker-  https://docs.docker.com/install/
```
Comment (rkawula) [inline README.md]: Start list with 1. for Markdown formatting.

File: README.md [hunk 2]
```diff
@@ -3,3 +3,18 @@
...
+1. Copy application.properties.example to an application.properties file in the same directory
+2. Replace the password <Replace Me!> in both Dockerfile and the new application.properties file with any secure password of your choosing... or don't, it's just a dev environment.
+3. Run these commands from the shell in the base api directory
+ 3a. docker build -t mysql .
```
Comment (rkawula) [inline README.md]: Check markdown formatting for sub-bullet points.

File: pom.xml [hunk 3]
```diff
@@ -18,9 +18,20 @@
 	</parent>
 
 	<dependencies>
+		<dependency>
+		<groupId>org.springframework.boot</groupId>
```
Comment (rkawula) [inline pom.xml]: Make indentation consistent.

File: src/main/java/com/band/api/Application.java [hunk 4]
```diff
@@ -2,8 +2,9 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 
-
+@EnableJpaRepositories(basePackages = "com.band.api.repository")
```
Comment (rkawula) [inline src/main/java/com/band/api/Application.java:7]: 👍

File: src/main/java/com/band/api/repository/BandSkillsNeeded.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+@Entity
+@Table(name = "band_skills_needed", catalog = "band")
+public class BandSkillsNeeded {
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    private Integer id;
+
+    @Column(nullable = false)
+    private Integer band_id;
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/BandSkillsNeeded.java]: Make whitespace formatting between fields consistent.

File: src/main/java/com/band/api/repository/BandSkillsNeeded.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    @Column(columnDefinition = "VARCHAR(128)", nullable = false)
+    private String skill;
+    @Column( nullable = false)
+    private Integer experience;
+    public BandSkillsNeeded() {
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/BandSkillsNeeded.java]: Don't need 0-param constructor.

File: src/main/java/com/band/api/repository/Skill.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+@Entity
+@Table(name = "skill", catalog = "band")
+public class Skill {
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    private Integer id;
+
+    @Column(columnDefinition = "VARCHAR(128)", nullable = false)
+    private String skill;
+    public Skill() {
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/Skill.java]: Don't need 0-param constructor; let's talk offline about Builder patterns.

File: src/main/java/com/band/api/repository/User.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Column(columnDefinition = "VARCHAR(128)", nullable = false)
+    private String state;
+
+    @Column(columnDefinition = "VARCHAR(128)", nullable = false)
+    private String city;
+
+
+
+    public User() {
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/User.java]: Remove.

File: src/main/java/com/band/api/repository/User.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    public String getUsername() {
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/User.java]: Remove all of the getters & setters -- read about lombok https://projectlombok.org/features/GetterSetter

Can use @Getter @Setter annotation at the field level or classes level, and can use @Data to apply both getter and setter.

File: src/main/java/com/band/api/repository/User.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public String getPassword_hash() {
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/User.java]: No _ in names in Java.

File: src/main/java/com/band/api/repository/Band.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
+package com.band.api.repository;
+
+import javax.persistence.*;
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/Band.java]: In general don't do star imports, remove this line and add each item in one at a time (for all repository files).

**Review by rkawula — CHANGES_REQUESTED** [review]
See comments.

File: src/main/java/com/band/api/controller/SignUpController.java [hunk 12]
```diff
@@ -0,0 +1,4 @@
+package com.band.api.controller;
+
+public class SignUpController {
```
Comment (rkawula) [inline src/main/java/com/band/api/controller/SignUpController.java]: Seems unrelated to this feature -- remove from this commit.

File: src/main/java/com/band/api/repository/BandMember.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Id
+    @Column(columnDefinition = "VARCHAR(128)")
+    private String role;
+
+    @Column(columnDefinition = "DATE", nullable = false)
+    private Date joined_date;
+
+
+    public BandMember() {
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/BandMember.java]: Remove.

File: src/main/java/com/band/api/repository/Band.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "band", catalog = "band")
+public class Band {
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/Band.java]: Rename all repositories to "...Repository" to make it easier down the line.

File: src/main/java/com/band/api/repository/BandMember.java [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+@Entity
+@Table(name = "band_member", catalog = "band")
+public class BandMember implements Serializable {
+    @Id
+    @Column(nullable = false)
+    private Integer band_id;
+
+    @Id
+    @Column(nullable = false)
+    private Integer member_id;
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/BandMember.java]: Replace underscore field names with camel case.

File: src/main/java/com/band/api/service/SignUpService.java [hunk 16]
```diff
@@ -0,0 +1,4 @@
+package com.band.api.service;
+
+public class SignUpService {
```
Comment (rkawula) [inline src/main/java/com/band/api/service/SignUpService.java]: Remove unrelated changes from commit.

**Review by rkawula — CHANGES_REQUESTED** [review]

File: src/main/java/com/band/api/repository/BandMemberDao.java [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+import javax.persistence.Table;
+
+import java.io.Serializable;
+import java.util.Date;
+
+@Entity
+@Builder
+@Getter
+@Table(name = "band_member", catalog = "band")
+public class BandMemberDao implements Serializable {
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/BandMemberDao.java:18]: Should all of the DAO classes implement Serializable or why only this one?

File: src/main/java/com/band/api/repository/UserEndorsementDao.java [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Builder
+@Getter
+@Table(name = "user_endorsement", catalog = "band")
+public class UserEndorsementDao {
+    @Id
```
Comment (rkawula) [inline src/main/java/com/band/api/repository/UserEndorsementDao.java:16]: Does this need a generation strategy?

**Review by rkawula — CHANGES_REQUESTED** [review]

File: src/main/java/com/band/api/repository/BandMemberDao.java (hunk 17 above)
Comment (TheLastKiwi) [reply src/main/java/com/band/api/repository/BandMemberDao.java:18]: Only tables with composite primary keys need to implement Serializable.

File: src/main/java/com/band/api/repository/UserEndorsementDao.java (hunk 18 above)
Comment (TheLastKiwi) [reply src/main/java/com/band/api/repository/UserEndorsementDao.java:16]: Probably, but none of these tables are connected as is so there are a lot of other problems too. They're just templated right now. If this is the only problem we have other problems.

**Review by rkawula — APPROVED** [review]
